common/src/stack/command/stack/commands/load/json/__init__.py

```diff
@@ -0,0 +1,141 @@
+# @copyright@
+# Copyright (c) 2006 - 2018 Teradata
+# All rights reserved. Stacki(r) v5.x stacki.com
+# https://github.com/Teradata/stacki/blob/master/LICENSE.txt
+# @copyright@
+#
+# @rocks@
+# Copyright (c) 2000 - 2010 The Regents of the University of California
+# All rights reserved. Rocks(r) v5.4 www.rocksclusters.org
+# https://github.com/Teradata/stacki/blob/master/LICENSE-ROCKS.txt
+# @rocks@
+
+import stack.commands
+import json
+import sys
+import os
+from stack.exception import CommandError
+import subprocess
+import logging
+import stack.commands
+
+class command(stack.commands.load.command, stack.commands.Command):
+	MustBeRoot = 0
+	notifications = True
+
+
+class Command(command):
+	"""
+	Load configuration data from the provided json document. If no arguments
+	are provided then all plugins will be run.
+
+	<param optional='0' type='string' name='file'>
+	The json document containing the configuration information.
+	</param>
+
+	<arg optional='1' type='string' name='software'>
+	Load pallet, cart, and box data. If the pallet exists on a remote
+	server that requires authentication, be sure to provide the username
+	and password in the corresponding json fields.
+	</arg>
+
+	<arg optional='1' type='string' name='host'>
+	Load name, rack, rank, interface, attr, firewall, box, appliance,
+	comment, metadata, environment, osaction, route, group, partition,
+	and controller data for each host.
+	</arg>
+
+	<arg optional='1' type='string' name='network'>
+	Load name, address, gateway, netmask, dsn, pxe, mtu, and zone
+	data for each network.
+	</arg>
+
+	<arg optional='1' type='string' name='global'>
+	Load attr, route, firewall, partition, and controller data for
+	the global scope.
+	</arg>
+
+	<arg optional='1' type='string' name='os'>
+	Load name, attr, route, firewall, partition, and controller
+	data for each os.
+	</arg>
+
+	<arg optional='1' type='string' name='appliance'>
+	Load name, attr, route, firewall, partition, and controller
+	data for each appliance.
+	</arg>
+
+	<arg optional='1' type='string' name='group'>
+	Load name data for each group.
+	</arg>
+
+	<arg optional='1' type='string' name='bootaction'>
+	Load name, kernel, ramdisk, type, arg, and os data for each
+	bootaction.
+	</arg>
+	"""
+
+	def checks(self, args):
+
+		# Make sure that the user is not attepting to add more than one frontend
+		if ('host' in args or not args) and 'host' in self.data:
```

Comment:
Check 'args' before checking for membership `in` it.

Reply:
When no args are given, args get passed through as an empty list, so this doesn't crash. Should it still be changed anyway just for readability?

Reply:
I'd change it for readability but also future proofing.  It's just safer syntax.